internal/controller/rbac/provider/binding/reconciler.go

```diff
@@ -169,27 +169,30 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 		return reconcile.Result{Requeue: false}, nil
 	}
 
-	l := &corev1.ServiceAccountList{}
+	l := &appsv1.DeploymentList{}
 	if err := r.client.List(ctx, l); err != nil {
```

Comment:
We could probably filter this down to the namespace where we expect runtime deployments to exist (i.e. `crossplane-system`). Not a blocker though - this would be true for the previous service account based implementation too.